packages/tinacms/src/admin/components/GetCollection.tsx

```diff
@@ -32,8 +31,10 @@ export const useGetCollection = (
   const [resetState, setResetSate] = useState(0)
 
   useEffect(() => {
+    let cancelled = false
```

Comment:
This makes sense but just curious why it's necessary in the scope of this issue? Are we not handling a race condition properly somewhere?

Reply:
I think it probably snuck in when this feature (the redirect to the edit page of a single item collection) was first implemented. There is a `Can't perform a React state update on an unmounted component` warning in the console without this, because I think it is still executing when it navigates from the collection list url to the edit url. I thought it would be good to clean up here along with this.

Reply:
Ah great, I've seen that error running around too.